src/modm/board/disco_f411ve/board.hpp

```diff
@@ -0,0 +1,231 @@
+/*
+ * Copyright (c) 2024, Thomas Sommer
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+#pragma once
+
+#include <modm/architecture/interface/clock.hpp>
+#include <modm/driver/inertial/lis3dsh.hpp>
+#include <modm/platform.hpp>
+
+using namespace modm::platform;
+
+namespace Board
+{
+/// @ingroup modm_board_disco_f411ve
+/// @{
+using namespace modm::literals;
+
+/// STM32F411 running at 100MHz generated from the external 8MHz crystal
+struct SystemClock
+{
+	static constexpr uint32_t Frequency = 84_MHz;
+	static constexpr uint32_t Ahb = Frequency;
+	static constexpr uint32_t Apb1 = Frequency / 2;
+	static constexpr uint32_t Apb2 = Frequency;
+
+	static constexpr uint32_t Adc = Apb2;
+
+	static constexpr uint32_t Spi1 = Apb2;
+	static constexpr uint32_t Spi2 = Apb1;
+	static constexpr uint32_t Spi3 = Apb1;
+	static constexpr uint32_t Spi4 = Apb2;
+	static constexpr uint32_t Spi5 = Apb2;
+
+	static constexpr uint32_t Usart1 = Apb2;
+	static constexpr uint32_t Usart2 = Apb1;
+	static constexpr uint32_t Usart3 = Apb1;
+	static constexpr uint32_t Uart4 = Apb1;
+	static constexpr uint32_t Uart5 = Apb1;
+	static constexpr uint32_t Usart6 = Apb2;
+	static constexpr uint32_t Uart7 = Apb1;
+	static constexpr uint32_t Uart8 = Apb1;
+
+	static constexpr uint32_t I2c1 = Apb1;
+	static constexpr uint32_t I2c2 = Apb1;
+	static constexpr uint32_t I2c3 = Apb1;
+
+	static constexpr uint32_t Apb1Timer = Apb1 * 2;
+	static constexpr uint32_t Apb2Timer = Apb2 * 2;
+	static constexpr uint32_t Timer1 = Apb2Timer;
+	static constexpr uint32_t Timer2 = Apb1Timer;
+	static constexpr uint32_t Timer3 = Apb1Timer;
+	static constexpr uint32_t Timer4 = Apb1Timer;
+	static constexpr uint32_t Timer5 = Apb1Timer;
+	static constexpr uint32_t Timer9 = Apb2Timer;
+	static constexpr uint32_t Timer10 = Apb2Timer;
+	static constexpr uint32_t Timer11 = Apb2Timer;
+
+	static constexpr uint32_t Usb = 48_MHz;
```

Comment:
Why don't we move the `const Rcc::PllFactors pllFactors` above final clocks and calculate them? Constraints like the required 48Mhz may be static asserted then!

Something like this:
```
static constexpr HSE_crystal = 8;
static constexpr uint32_t Usb = HSE_crystal / pllM * pllN / pllQ;
static_assert(Usb == 48, "Usb clock must be 48Mhz!");
```

Reply:
Ohhhhhhh, that's an **excellent** idea!!! I'll merge this first, then prototype it. We can probably also put the AHB/APB prescalers in there.

Reply:
Something easy, to build up 'momentum of success' will also support my motivational chemistry to complete the old graphic related PRs.

Reply:
see #1137